qemu/tests/blockdev_commit_specify_node.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from virttest.qemu_monitor import QMPCmdError
+
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitSpecifyNode(BlockDevCommitTest):
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["base-node", "top-node", "speed"]
+        arguments = self.params.copy_from_keys(options)
+        if self.params.get("bst_qmp_error_msg"):
+            arguments["base-node"] = self.get_node_name(snapshot_tags[-2])
+            arguments["top-node"] = self.get_node_name(snapshot_tags[-2])
+            device = self.get_node_name(snapshot_tags[-1])
+            expected_error_msg = self.params.get("bst_qmp_error_msg")
+        if self.params.get("ptcb_qmp_error_msg"):
+            arguments["base-node"] = self.get_node_name(snapshot_tags[-2])
+            arguments["top-node"] = self.get_node_name(snapshot_tags[0])
+            device = self.get_node_name(snapshot_tags[-1])
+            expected_error_msg = self.params.get("ptcb_qmp_error_msg")
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        try:
+            self.main_vm.monitor.cmd(cmd, args)
+        except QMPCmdError as e:
+            logging.info("Error message is %s", e.data)
+            if expected_error_msg not in str(e.data):
+                self.test.fail("Error message not as expected")
+        else:
+            self.test.fail("Block commit should fail with %s,but "
+                           "block commit succeeded unexpectedly"
+                           % expected_error_msg)
+
+    def run_test(self):
+        self.pre_test()
+        try:
+            self.commit_snapshots()
+        finally:
+            self.post_test()
+
+
+def run(test, params, env):
+    """
+    Block commit with based and top is the same
```

Comment:
It could be better that both test scenarios are described here

Reply:
Thanks all your suggestions,updated the code.